io/net/net_tools.py

```diff
@@ -0,0 +1,289 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Basheer K<basheer@linux.vnet.ibm.com>
+#
+# Based on code by  Hong Bo Peng <penghb@cn.ibm.com>
+# copyright: 2003, 2015 IBM Corp
+
+import os
+import re
+from avocado import Test
+from avocado import main
+from avocado.core import data_dir
+from avocado.utils import process, distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class net_tools(Test):
+    """
+    Net_tools class verify the functionality of various network tools.
+    hostname: set/get hostname by cmd. Check it with the expected result.
+    ifconfig: setup an alias for lo. Check it can be listed.
+    netstat: run it with different options. Check the return code.
+    arp: run it with different options. Check the return code.
+    traceroute,traceroute6: run for localhost. It should report 1 hop.
+    route: run it with -n. Check the return code.
+    ipmaddr: run it and check the return code.
+    iptunnel: create sit1 and check it can be list. Then remove it and
+              check it is removed from the list.
+    """
+
+    def setUp(self):
+        """
+        Verify Required tools are installed or not.
+        """
+        if os.getuid() != 0:
+            exit("You need to have root privileges to execute this script. \
+                 \nPlease try again, using 'sudo'. Exiting.")
+        sm = SoftwareManager()
+        self.detected_distro = distro.detect()
+        if self.detected_distro.name == "SuSE":
+            net_tools = ("net-tools", "traceroute")
+        else:
+            net_tools = ("net-tools", "hostname", "traceroute")
+        for pkg in net_tools:
+            if not sm.check_installed(pkg) and not sm.install(pkg):
+                self.error("%s package is need to test" % pkg)
+        self.ipv6 = False
+        if os.path.exists("/proc/net/if_inet6"):
+            self.ipv6 = True
+        interface_out = process.system_output("ip route show default",
+                                              verbose=False,
+                                              shell=True)
+        search_obj = re.search(
+            "default via\s+(\S+)\s+dev\s+(\w+)", interface_out)
+        active_iface = search_obj.group(2)
+        self.default_router = search_obj.group(1)
+        if active_iface is None:
+            self.error("can't find active network interface")
+        tmpdir = data_dir.get_tmp_dir()
```

Comment:
This is not suppose to be used directly from tests. You can use the `self.workdir`